Improve documentation on how to setup a dev environment (#749)
getindiekit/indiekit: docs/development.md

```yml
version: '3.9'
services:
  mongo:
    container_name: mongo
    environment:
      - MONGO_INITDB_ROOT_USERNAME
      - MONGO_INITDB_ROOT_PASSWORD
    image: mongo:7.0.11
    network_mode: bridge
    ports:
    - '27018:27017'
    restart: always
    volumes:
      - mongo-data:/data/db
volumes:
  mongo-data:
    external: true
```

This particular configuration tells Docker to run the `mongo` service on port 27017 of the `mongo` container, and exposes port 27018 on the Docker host (e.g. your laptop).

You can set the necessary environment variables in a `.env` file, or use an `.envrc` file that will be automatically loaded by [direnv](https://direnv.net/) when
you enter the root of the project.

```txt
export MONGO_INITDB_ROOT_USERNAME="user"
export MONGO_INITDB_ROOT_PASSWORD="password"
export MONGO_URL="mongodb://$MONGO_INITDB_ROOT_USERNAME:$MONGO_INITDB_ROOT_PASSWORD@localhost:27018"

```

> [!TIP]
> If you want to inspect data stored in a MongoDB database, you can use the [MongoDB shell](https://www.mongodb.com/products/tools/shell) or a GUI like [Compass](https://www.mongodb.com/products/tools/compass).

> ## Configure the content store

Author: Oh cool, I didn't know it.

Indiekit performs CRUD operations on files. These files needs to be stored in a so-called content store. Different content stores require different configurations and credentials.

> ### GitHub content store

Author: Maybe I would rephrase the paragraph like this:

> Indiekit performs CRUD operations on some kind of persistence storage: a filesystem, a git repository, or an object storage like AWS S3. The so-called Indiekit content stores provide an implementation of these CRUD operations.
> Different content stores require different configurations and credentials.

As for testing, I would consider both the filesystem and GitHub as external services to Indiekit, so those tests would be integration tests. I wouldn't add anything more about testing in this document. Maybe I would add some guidelines in `contributing.md` (e.g. if you write a function, write some unit tests for it; if you write a syndicator, write a few integration tests for it).

If you want to use a GitHub repository as your content store, you will need to provide a [GitHub personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens). If you create a [fine-grained personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#fine-grained-personal-access-tokens), give it **Read and write** access to **Contents**, and set a reasonable expiration (e.g. 90 days).

```txt
export GITHUB_USER="johndoe"
export GITHUB_REPO="indiekit-content"
export GITHUB_BRANCH="main"
export GITHUB_TOKEN="github_pat_*****"
```

> ## Configure the syndicators

Reviewer: I think this can be an optional step, as only relevant if you want to test syndicating posts.

```suggestion
## Configure a syndicator (optional)
```

In order to tell Indiekit to [share content with other third-party websites](introduction#sharing-content-with-third-party-websites-syndication), you need to configure one or more syndicators.

> ### Mastodon syndicator

Author: As far as I know, the configuration for the Mastodon syndicator stays the same, no matter the user wants to just use Indiekit, or contribute the the syndicator itself.

So we could add something along these lines:

> To know all the configuration options for this syndicator, refer to its README.md

If you want to syndicate your content to Mastodon, you will need to provide a Mastodon access token with **read and write** access. You can generate an access token using [this web app](https://takahashim.github.io/mastodon-access-token/), or making a POST request as explained [on this page of the Mastodon documentation](https://docs.joinmastodon.org/client/token/).

> [!WARNING]
> As far as I know, [Mastodon access tokens do not expire](https://mastodon.social/@kevinhooke/109377838604407902).

```txt
export MASTODON_ACCESS_TOKEN="*****"
export MASTODON_USER="johndoe"
export MASTODON_URL="https://fosstodon.org"
```

## Running Indiekit locally

### Installation

Install all dependencies:

```sh
npm install
```

> ### Environment variables

Reviewer: As mentioned, maybe this guidance should appear earlier in this document? Perhaps using `PUBLICATION_URL` as an example variable to set?

Happy to mention direnv, but the default means of setting variables should be using dotenv (perhaps to later be replaced with guidance that refers to Node’s native support for reading `.env` files).

Update your `.env` / `.envrc` file with the environment variables required by the Indiekit server, the MongoDB database, the Indiekit content store and the Indiekit syndicators.

```txt
export PUBLICATION_URL="https://example.com"

# Used by @indiekit/endpoint-auth to sign and verify tokens and salt password
export SECRET="*****"

# Hashed and salted password used when signing in. You can generate this value by visiting /auth/new-password
export PASSWORD_SECRET="="*****"

# Environment variables for MongoDB
# Environment variables for Indiekit content store (e.g. GitHub)
# Environment variables for Indiekit syndicators (e.g. Mastodon)
```

### Run Indiekit

Start the server:

```sh
npm start
```

> In alternative, automatically restart the server whenever a file change is detected:

Reviewer: ```suggestion
Alternatively, you restart the server whenever a file change is detected:
```
